pkg/pgmodel/sql_ingest.go

```diff
@@ -312,6 +316,22 @@ func (p *pgxInserter) getMetricInserter(metric string) chan insertDataRequest {
 	return inserter.(chan insertDataRequest)
 }
 
+type noOpInserter struct{}
+
+func (*noOpInserter) InsertNewData(rows map[string][]samplesInfo) (uint64, error) {
+	return 0, noOpIngestError
```

Comment:
Hmm. If I understand correctly this error will return an HTTP 500 response? The problem is Prometheus will classify this as a Recoverable error(https://github.com/prometheus/prometheus/blob/master/storage/remote/client.go#L200)  and will retry (https://github.com/prometheus/prometheus/blob/master/storage/remote/queue_manager.go#L921).

But I don't think we want retrying here? 

I think a status 400 0r 404 is more appropriate here.

Honestly this may be better done at the mux level when setting up routing rather than thru a no-op inserter. But I'll leave that up to you.

Reply:
AFAIK, there is plan to make 4xx errors as recoverable errors and just the 5xx as unrecoverable. at least, that's what the discussion says on the link I mentioned below (I knew that as I am working on that issue)

related issue: https://github.com/prometheus/prometheus/issues/7078

Reply:
@Harkishen-Singh thats for rate limiting errors it seems (429 code). Will change this to do 404 Not Found on write in read-mode. That one shouldn't be recoverable IMO.

Reply:
yeah making 404 recoverable would just be wrong